From bbc01115577a3e892452a17b7b27590519fcdf56 Mon Sep 17 00:00:00 2001
From: "Nalla, Pradeep" <pnalla@marvell.com>
Date: Sat, 10 Oct 2020 16:39:15 +0530
Subject: [PATCH 19/19] octeontx2-npa: Fix NPA PF driver commit.

commit 71bb9befa22a3b0a1d3b33b4b824ad597e02f6b4 from
git@git.assembla.com:cavium/WindRiver.linux.git

A header file advertising the API exported by NPA PF driver introduced
in the commit "8465a16348e9", is missed in that commit. This made the
driver unusable by components that need this driver. Added missing file
in this commit and h/w units that need pool manager can simply include
this file and use the API.

Fixes: 8465a16348e9("soc: octeontx2: Add NPA PF driver")
Change-Id: Iaba6e4303fdc2fab3fae2ce506ec83b7ccccc87d
Signed-off-by: Nalla, Pradeep <pnalla@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/37686
Reviewed-by: Radha Chintakuntla <radhac@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Abhijit Ayarekar <aayarekar@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/49095
Reviewed-by: Yuri Tolstov <Yuri.Tolstov@cavium.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/soc/marvell/octeontx2-npa/npa.c     |  1 +
 drivers/soc/marvell/octeontx2-npa/npa_api.h | 19 +++++++++++++++++++
 2 files changed, 20 insertions(+)
 create mode 100644 drivers/soc/marvell/octeontx2-npa/npa_api.h

diff --git a/drivers/soc/marvell/octeontx2-npa/npa.c b/drivers/soc/marvell/octeontx2-npa/npa.c
index 5c1ff9f33a10..df9bbb0f06e5 100644
--- a/drivers/soc/marvell/octeontx2-npa/npa.c
+++ b/drivers/soc/marvell/octeontx2-npa/npa.c
@@ -20,6 +20,7 @@
 #include "rvu_reg.h"
 #include "mbox.h"
 #include "npa.h"
+#include "npa_api.h"
 
 #define DRV_NAME "octeontx2-npapf"
 #define DRV_VERSION "1.0"
diff --git a/drivers/soc/marvell/octeontx2-npa/npa_api.h b/drivers/soc/marvell/octeontx2-npa/npa_api.h
new file mode 100644
index 000000000000..559f175ad360
--- /dev/null
+++ b/drivers/soc/marvell/octeontx2-npa/npa_api.h
@@ -0,0 +1,19 @@
+/* SPDX-License-Identifier: (GPL-2.0)
+ * OcteonTX2 NPA driver
+ *
+ * Copyright (C) 2020 Marvell International Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+/* Initializa aura pool pair */
+int npa_aura_pool_init(int pool_size, int buf_size, u32 *aura_handle,
+		       struct device *owner);
+/* Teardown aura pool pair */
+int npa_aura_pool_fini(const u32 aura_handle, struct device *owner);
+u64 npa_alloc_buf(u32 aura);
+void npa_free_buf(u32 aura, u64 buf);
+/* Get PF function used for aura */
+u16 npa_pf_func(u32 aura);
-- 
2.31.1

